eclipse-theia/theia-cpp-extensions
# repo: cleanup repository and support node 12

#### What it does


The pull-request includes the following changes:
- [x] removes the deprecated `@theia/cpp` extension from the build
- [x] removes deprecated `@theia` extensions (`languages` and `json`)
- [x] updates the repository to support `node 12` which is the supported version from the framework
- [x] removes the UI test-suite (outdated and dependencies do not work with `node 12`)
- [x] adds `vscode-clangd` to the examples (in replacement for `@theia/cpp`)

#### How to test


- CI should now be green again (build and test)
- the `example/electron` app should successfully start
- the `example/browser` app should successfully start

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>
